### PR TITLE
Additionally order by primary key if implicit_order_column is not uniq

### PR DESCRIPTION
**Pull request by pawurb**
As mentioned here https://github.com/rails/rails/pull/34480/files#r234781773 the current implementation of `implicit_order_column` does not guarantee that the objects will always be returned in the same order if `implicit_order_column` contains duplicate values. 

This patch adds a secondary sort by primary key (if it's not set as a `implicit_order_column` already) to guarantee the consistent output.

### CODE REVIEW

**Comment by pawurb** [comment]
The reasoning behind this patch is that with UUID used as a primary key setting `created_at` as `implicit_order_column` can become a common practice.  Batch created objects will all have the same `created_at` value, causing the inconsistent behaviour for`last` and `first` methods.

File: activerecord/lib/active_record/relation/finder_methods.rb
Comment (eugeneius) [inline activerecord/lib/active_record/relation/finder_methods.rb]:
```suggestion
          if implicit_order_column && primary_key && implicit_order_column != primary_key
```

If a model has no primary key and has `implicit_order_column` configured, we shouldn't add the secondary sort.

Both cases need a test.

Comment (pawurb) [reply activerecord/lib/active_record/relation/finder_methods.rb]: @eugeneius I've added tests covering the cases I could think of.

File: activerecord/test/cases/finder_test.rb
Comment (eugeneius) [inline activerecord/test/cases/finder_test.rb]: These lines could be shorter and more readable if the table names and column names were quoted together:

```suggestion
    assert_sql(/ORDER BY #{Regexp.escape(c.quote_table_name("topics.title"))} DESC, #{Regexp.escape(c.quote_table_name("topics.id"))} DESC LIMIT/i) {
```

It's admittedly a little strange to quote a column name with the `quote_table_name` method, but it works, and we already rely on it elsewhere:

https://github.com/rails/rails/blob/98754de1412870d7dae9eba1c7bac944b9b90093/activerecord/test/cases/batches_test.rb#L149
https://github.com/rails/rails/blob/98754de1412870d7dae9eba1c7bac944b9b90093/activerecord/test/cases/calculations_test.rb#L143

File: activerecord/lib/active_record/model_schema.rb
Comment (eugeneius) [inline activerecord/lib/active_record/model_schema.rb]:
```suggestion
    # auto-incrementing integer, for example when it's a UUID. Records are subsorted
```

File: activerecord/lib/active_record/model_schema.rb
Comment (eugeneius) [inline activerecord/lib/active_record/model_schema.rb]:
```suggestion
    # by the primary key if it exists to ensure deterministic results.
```

**Review by eugeneius — COMMENTED** [review]
Could you add a changelog entry? Otherwise this is looking good 👍

**Review by eugeneius — APPROVED** [review]
LGTM, but this will need a committer's review before it can be merged.

**Comment by kaspth** [comment]
Sweet! Thanks for the fix @pawurb and the review @eugeneius ❤️

**Pull request merged**
